components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2397,7 +2398,25 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         const user = await this.checkUser("identifyUser");
         const email = getPrimaryEmail(user);
         const ideConfig = await this.ideService.getIDEConfig({ user: { id: user.id, email } });
-        return ideConfig.ideOptions;
+
+        const updatedOptions: Record<string, IDEOption> = {};
+        for (const [key, value] of Object.entries(ideConfig.ideOptions.options)) {
+            const opt = filter(value, (key) => key !== "versions") as IDEOption;
```

Comment:
(nit) I wonder is not it just
```js
const opt = {...value};
delete opt["versions"]
opt["pinnable"] = !!value.versions && value.versions.length > 0;
```

Reply:
mmm it will one need more `as any` casting for delete as the two type are not overlapped

Reply:
it is nit, nothing to change